tests/js/spec/views/alerts/utils.spec.jsx

```diff
@@ -81,4 +82,69 @@ describe('Alert utils', function () {
       });
     });
   });
+
+  describe('getQuerySource', () => {
+    it('should parse event type error or default', () => {
+      expect(getQueryDatasource('event.type:default OR event.type:error')).toEqual({
+        source: Datasource.ERROR_DEFAULT,
+        query: '',
+      });
+      expect(
+        getQueryDatasource(
+          'event.type:error OR event.type:default transaction.duration:<30s'
+        )
+      ).toEqual({
+        source: Datasource.ERROR_DEFAULT,
+        query: 'transaction.duration:<30s',
+      });
+      expect(
+        getQueryDatasource('event.type:error OR (event.type:default event.level:fatal)')
+      ).toEqual({
+        source: Datasource.ERROR_DEFAULT,
+        query: 'event.level:fatal)',
```

Comment:
I guess this query is pretty hard to parse, but is the unmatched parentheses expected. should we do any additional validation/cleaning steps after parsing out the datasource?

Reply:
yeah expected for this version